packages/roosterjs-editor-core/lib/coreAPI/insertNode.ts

```diff
@@ -87,8 +99,19 @@ export const insertNode: InsertNode = (core: EditorCore, node: Node, option: Ins
 
             // Final check to see if the inserted node is a block. If not block and the ask is to insert on new line,
             // add a DIV wrapping
-            if (insertedNode && option.insertOnNewLine && !isBlockElement(insertedNode)) {
-                wrap(insertedNode);
+            if (insertedNode && option.insertOnNewLine) {
+                let start = Array.isArray(insertedNode) ? insertedNode[0] : insertedNode;
+                // Check the last node if there're more than 1 node in list
+                let end =
+                    Array.isArray(insertedNode) && insertedNode.length > 1
+                        ? insertedNode[insertedNode.length - 1]
+                        : null;
+                if (start && !isBlockElement(start)) {
+                    wrap(start);
+                }
+                if (end && !isBlockElement(end)) {
+                    wrap(end);

```

Comment:
Actually, this may cause the inserted content looks different than expected.
For example:
```html
<span>Span1</span><span>Span2</span><span>Span3</span>
```

I expect the inserted content looks like:
Span1Span2Span3

But the actual result will be:
Span1
Span2
Span3

You don't need to wrap start and end node, but just need to wrap them together. And you can also make the check simpler:

```ts
if (insertNode) { // Need this check in case of empty document fragment
  const nodes = Array.isArray(insertedNode) ? insertedNode : [insertNode];
  if (!isBlockElement(nodes[0] || !isBlockElement(nodes[nodes.length - 1]) {
     wrap(nodes);
  }
}
```

Reply:
Got it. I misunderstood the logic of `wrap()`. Thanks for your clarification!